h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -70,7 +70,10 @@
   @Override public boolean haveMojo() { return true; }
 
   @Override protected void checkMemoryFootPrint() {
-    long mem_usage = 8 /*doubles*/ * (_parms._k * _train.numCols() + _parms._k );  // loose estimation of memory usage
+    HeartBeat hb = H2O.SELF._heartbeat;
+    double p = hex.util.LinearAlgebraUtils.numColsExp(_train,true);
+    long mem_usage = (long)(hb._cpus_allowed * p*_parms._k * 8 * Math.log((double)_train.lastVec().nChunks())/
```

Comment:
seems strange to be taking number of cpus from heartbeat. I think it's a member on H2O object or somewhere similar.